__tests__/commands/list.js

```diff
@@ -25,115 +25,153 @@ const runList = buildRun.bind(null, BufferReporter, fixturesLoc, (args, flags, c
+      const trees = [makeTree('is-plain-obj@1.1.0')];
 
-    rprtr.tree('list', trees);
+      rprtr.tree('list', trees);
 
-    expect(tree).toEqual(rprtr.getBuffer());
+      expect(tree).toEqual(rprtr.getBuffer());
+    });
   });
-});
 
-test.concurrent('accepts an argument', (): Promise<void> => {
-  return runList(['is-plain-obj'], {}, 'one-arg', (config, reporter): ?Promise<void> => {
-    const rprtr = new BufferReporter({});
-    const tree = reporter.getBuffer().slice(-1);
-    const trees = [makeTree('is-plain-obj@1.1.0')];
+  test.concurrent('matches exactly without glob', (): Promise<void> => {
+    return runList(['gulp'], {}, 'glob-arg', (config, reporter): ?Promise<void> => {
+      const rprtr = new BufferReporter({});
+      const tree = reporter.getBuffer().slice(-1);
+      const trees = [makeTree('gulp@3.9.1', {color: 'bold'})];
 
-    rprtr.tree('list', trees);
+      rprtr.tree('list', trees);
 
-    expect(tree).toEqual(rprtr.getBuffer());
+      expect(tree).toEqual(rprtr.getBuffer());
+    });
   });
-});
 
-test.concurrent('matches exactly without glob', (): Promise<void> => {
-  return runList(['gulp'], {}, 'glob-arg', (config, reporter): ?Promise<void> => {
-    const rprtr = new BufferReporter({});
-    const tree = reporter.getBuffer().slice(-1);
-    const trees = [makeTree('gulp@3.9.1', {color: 'bold'})];
+  test.concurrent('expands a glob', (): Promise<void> => {
+    return runList(['gulp*'], {}, 'glob-arg', (config, reporter): ?Promise<void> => {
+      const rprtr = new BufferReporter({});
+      const tree = reporter.getBuffer().slice(-1);
+      const trees = [makeTree('gulp@3.9.1', {color: 'bold'}), makeTree('gulp-babel@6.1.2', {color: 'bold'})];
 
-    rprtr.tree('list', trees);
+      rprtr.tree('list', trees);
 
-    expect(tree).toEqual(rprtr.getBuffer());
+      expect(tree).toEqual(rprtr.getBuffer());
+    });
   });
-});
 
-test.concurrent('expands a glob', (): Promise<void> => {
-  return runList(['gulp*'], {}, 'glob-arg', (config, reporter): ?Promise<void> => {
-    const rprtr = new BufferReporter({});
-    const tree = reporter.getBuffer().slice(-1);
-    const trees = [makeTree('gulp@3.9.1', {color: 'bold'}), makeTree('gulp-babel@6.1.2', {color: 'bold'})];
+  test('lists all dependencies when not production', (): Promise<void> => {
+    const nodeEnv = process.env.NODE_ENV;
+    process.env.NODE_ENV = 'development';
+    return runList([], {}, 'dev-deps-prod', (config, reporter): ?Promise<void> => {
+      const rprtr = new BufferReporter({});
+      const tree = reporter.getBuffer().slice(-1);
+      const children = [{name: 'is-plain-obj@^1.0.0', color: 'dim', shadow: true}];
+      const trees = [
+        makeTree('left-pad@1.1.3', {color: 'bold'}),
+        makeTree('sort-keys@1.1.2', {children, color: 'bold'}),
+        makeTree('is-plain-obj@1.1.0'),
+      ];
+
+      rprtr.tree('list', trees);
+
+      expect(tree).toEqual(rprtr.getBuffer());
```

Comment:
Here's my reasoning:

If you manually tested this and know this output is correct, we can safely assume that the output should not change unless the behavior changes. Thus, we can store this sample output as a reference and in this output we should/should not have devDependencies based on the prod state. Makes sense?